pygatt/backends/gatttool/gatttool.py

```diff
@@ -26,7 +26,7 @@
 if hasattr(bytes, 'fromhex'):
     # Python 3.
     def _hex_value_parser(x):
-        return bytearray.fromhex(x.decode('utf8'))
```

Comment:
@peplin I did verify this using real hardware that returned data, so the data I built and tested the proposed changes on came from `gatttool`. The problem was that I got a `bytes` string in Python 3 (the binary text type there), whereas in Python 2 I got a `str` (the binary text type there). The `bytesarray.fromhex` method accepted only `str` input in both environments, so I added a `decode` to make the `bytes` into a `str` in Python 3.
I believe that the `pexpect`/`gatttool` solution is consistent in the sense that it returns binary representations of string data in both environments, the only problem is that Python 3 is less adapted to use such representation in general.